src/CMakeLists.txt

```
@@ -46,9 +46,38 @@ PODIO_GENERATE_DICTIONARY(podioDict ${headers} SELECTION selection.xml
 # prevent generating dictionary twice
 set_target_properties(podioDict-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
 
+# check if the linker supports as needed or not
+set(STORE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+include(CheckCCompilerFlag)
+set(CMAKE_REQUIRED_FLAGS "-Wl,--no-as-needed")
+check_c_compiler_flag("" TEST_NO_AS_NEEDED)
+if(TEST_NO_AS_NEEDED)
+    message(STATUS "Linker supports: -Wl,--no-as-needed")
+else()
+    message(STATUS "Linker does not support: -Wl,--no-as-needed")
+endif()
+set(CMAKE_REQUIRED_FLAGS "-Wl,--as-needed")
+check_c_compiler_flag("" TEST_AS_NEEDED)
+if(TEST_AS_NEEDED)
+    message(STATUS "Linker supports: -Wl,--as-needed")
+else()
+    message(STATUS "Linker does not support: -Wl,--as-needed")
+endif()
+set(CMAKE_REQUIRED_FLAGS ${STORE_REQUIRED_FLAGS})
+
 target_sources(podioDict PRIVATE podioDict.cxx)
 # ROOT IO without the Dict does not work, so we link RootIO against the Dict
-target_link_libraries(podioRootIO PUBLIC podioDict)
+# need to disable as needed if linker supports it
+if(TEST_NO_AS_NEEDED AND TEST_AS_NEEDED)
+  target_link_libraries(podioRootIO
+    PRIVATE
+      -Wl,--no-as-needed podioDict -Wl,--no-as-needed
```

Of course I meant to write
```suggestion
      -Wl,--no-as-needed podioDict -Wl,--as-needed
```
But then it probably wouldn't work, because cmake re-arranges the order of flags and libraries...
So, just set no-as-needed and don't bother with resetting?
(push-state/pop-state has the same problem)

---

Nevermind, the problem was using PUBLIC instead of INTERFACE in the line below, then podioDict was twice in the list of libraries and de-duplication made it look like re-arranging was happening.